### ISSUE TITLE
The client failed to call mishards search, mishards reported connection error

### ISSUE THREAD

**Issue by Swidasya** [milvus-io/milvus#2493]
During real-time concurrent search, there will be search failures, specific related logs:

**error in client search:**
`Search RPC failed: Status{Code=UNKNOWN, description=Exception calling application: , cause:null}
`
**error in milvus error log:**
```
ERROR | default | [ENGINE] No file to search in file id list
2020-05-13 19:07:36,505 | ERROR | default | [SERVER] [search][0] Query fail: Invalid file id
2020-05-13 19:07:36,505 | ERROR | default | [SERVER] Request failed with code: Error: Invalid file id

```
**error in mishards log:**
```
2020-05-13 11:07:44,786 | ERROR | grpc._server | ThreadPoolExecutor-0_29: Exception calling application:  (_server.py:444)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/grpc/_server.py", line 434, in _call_behavior
    response_or_iterator = behavior(argument, context)
  File "/usr/local/lib/python3.6/site-packages/grpc_opentracing/grpcext/_interceptor.py", line 272, in adaptation
    _UnaryServerInfo(self._method), handler)
  File "/usr/local/lib/python3.6/site-packages/grpc_opentracing/_server.py", line 152, in intercept_unary
    response = handler(request, servicer_context)
  File "/usr/local/lib/python3.6/site-packages/grpc_opentracing/grpcext/_interceptor.py", line 268, in handler
    servicer_context)
  File "/source/mishards/server.py", line 70, in wrapper
    return func(*args, **kwargs)
  File "/source/mishards/service_handler.py", line 360, in Search
    metadata=metadata)
  File "/source/mishards/service_handler.py", line 170, in _do_query
    res.result()
  File "/usr/local/lib/python3.6/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/local/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/source/mishards/service_handler.py", line 137, in search
    conn = self.router.query_conn(addr, metadata=metadata)
  File "/source/mishards/router/__init__.py", line 23, in query_conn
    conn.on_connect(metadata=metadata)
  File "/source/mishards/connections.py", line 71, in on_connect
    metadata=metadata))
mishards.exceptions.ConnectionConnectError
```

**My env: Cluster mode**
1. mishards: 0.8.0
2. milvus:  0.8.0-gpu-d041520-464400
3. milvus-java-jdk: 0.7.0

**Application scenario:**
1. Multiple concurrent searches
2. In order to meet the real-time search, there may be multiple searches per second, each search volume within 5
3. All searches enjoy the same MilvusGrpcClient

**Comment by Swidasya** [comment]
>

**Comment by tinkerlin** [comment]
This problem already fixed on 0.9.0.
